### PR TITLE
Remove the remaining deprecated stuff

### PR DESCRIPTION
**Pull request by leofeyer**
This PR aims to remove all the remaining deprecated stuff, except for anything security related (will be handled in #4343) and `$GLOBALS['TL_LANGUAGE']` and `contao.upload_path`.

### CODE REVIEW

**Comment by leofeyer** [comment]
While working on this, I have noticed that the deprecations are inconsistent:

**We have deprecated**

* `BackendUser::isAllowed()`
* `BackendUser::canEditFieldsOf()`
* `FrontendUser::isMemberOf()`

**But we have not deprecated**

* `BackendUser::hasAccess()`
* `User::isMemberOf()`

@contao/developers Why is that?

File: core-bundle/src/Resources/contao/classes/BackendTemplate.php [hunk 1]
```diff
@@ -64,7 +64,7 @@ public function getResponse()
 	/**
 	 * Compile the template
 	 *
-	 * @internal Do not call this method in your code. It will be made private in Contao 5.0.
+	 * @internal
 	 */
 	protected function compile()
```
Comment (ausi) [inline core-bundle/src/Resources/contao/classes/BackendTemplate.php]: Don’t we need to make it `private` too?
```suggestion
	 * @internal
	 */
	private function compile()
```

File: core-bundle/src/Resources/contao/classes/FrontendTemplate.php [hunk 2]
```diff
@@ -81,7 +81,7 @@ public function getResponse($blnCheckRequest=false, $blnForceCacheHeaders=false)
 	 *
 	 * @throws UnusedArgumentsException If there are unused route parameters
 	 *
-	 * @internal Do not call this method in your code. It will be made private in Contao 5.0.
+	 * @internal
 	 */
 	protected function compile()
```
Comment (ausi) [inline core-bundle/src/Resources/contao/classes/FrontendTemplate.php]:
```suggestion
	 * @internal
	 */
	private function compile()
```

File: core-bundle/src/Resources/contao/library/Contao/Template.php [hunk 3]
```diff
@@ -480,7 +480,7 @@ public function param($strKey)
 	/**
 	 * Compile the template
 	 *
-	 * @internal Do not call this method in your code. It will be made private in Contao 5.0.
+	 * @internal
 	 */
 	protected function compile()
```
Comment (ausi) [inline core-bundle/src/Resources/contao/library/Contao/Template.php]:
```suggestion
	 * @internal
	 */
	private function compile()
```

File: core-bundle/src/Resources/contao/classes/BackendTemplate.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/classes/BackendTemplate.php]: Changed in e9b44906d07c6ff29a56be79b873d3d0d1f8e9e5.

File: core-bundle/src/Resources/contao/classes/FrontendTemplate.php [hunk 4]
```diff
@@ -81,9 +83,9 @@ public function getResponse($blnCheckRequest=false, $blnForceCacheHeaders=false)
 	 *
 	 * @throws UnusedArgumentsException If there are unused route parameters
 	 *
-	 * @internal Do not call this method in your code. It will be made private in Contao 5.0.
+	 * @internal
```
Comment (ausi) [inline core-bundle/src/Resources/contao/classes/FrontendTemplate.php]: This can be removed now ☺️

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/Resources/contao/classes/FrontendTemplate.php (hunk 4 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/classes/FrontendTemplate.php]: Of course. 🤦‍♂️ Removed in 9091327c48588f977cdf089fd94a7c3ffaefd574.

**Review by ausi — APPROVED** [review]

**Pull request merged**